### PR TITLE
AArch64: Use vector eor instructions for register exchange

### PR DESCRIPTION
**Pull request by Akira1Saitoh**
In register assignment, a temporary register is used for
exchanging floating point or vector registers.
However, if a register dependency has all 32 FPRs, the register assigner
cannot find the temporary register and fails abruptly.
Since vector eor instructions can be used for exchanging floating point
or vector registers without a temporary register, this commit changes
the register assigner to generate those instructions for exchanging registers
and removes the use of a temporary register.

Signed-off-by: Akira Saitoh <saiaki@jp.ibm.com>

### CODE REVIEW

File: compiler/aarch64/codegen/OMRMachine.cpp [hunk 1]
```diff
@@ -771,16 +770,14 @@ void OMR::ARM64::Machine::coerceRegisterAssignment(TR::Instruction *currentInstr
...
-
       if (targetRegister->getState() == TR::RealRegister::Blocked)
          {
 #ifdef DEBUG
          if (comp->getOption(TR_TraceCG))
             diagnostic(", which is blocked and assigned to %s",
                        currentTargetVirtual->getRegisterName(comp));
 #endif
-         if (!currentAssignedRegister || needTemp)
+         if (!currentAssignedRegister)
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRMachine.cpp:780]: Please merge the `if` block from L.781 to L.791 with the `else` block from L.803 to L.823.

**Comment by knn-k** [comment]
I found `coerceRegisterAssignment()` for Power has a variable `currentAssignedUseful`.  Introducing it might improve the register assignment for AArch64, but I think that can be done in another PR.

https://github.com/eclipse/omr/blob/ff6a49823dd5432795c5baa956dc87e0980dac07/compiler/p/codegen/OMRMachine.cpp#L1178-L1179

File: compiler/aarch64/codegen/OMRMachine.cpp (hunk 1 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRMachine.cpp]: Merged the blocks.

**Comment by knn-k** [comment]
Jenkins build aarch64

**Comment by knn-k** [comment]
The test failure with x86-64 macOS is independent of this PR.

**Comment by knn-k** [comment]
Tested with the following OpenJ9 test sets on internal servers: sanity.functional, extended.functional, sanity.system, sanity.openjdk

**Review by knn-k — APPROVED** [review]

**Pull request merged**